src/main/java/net/neoforged/neoforge/energy/EnergyStorage.java

```diff
@@ -22,6 +27,18 @@ public class EnergyStorage implements IEnergyStorage, INBTSerializable<Tag> {
     protected int maxReceive;
     protected int maxExtract;
 
+    public static final Codec<EnergyStorage> CODEC = RecordCodecBuilder.create(i -> i.group(
+        Codec.INT.fieldOf("capacity").forGetter(x -> x.capacity),
+        Codec.INT.fieldOf("maxReceive").forGetter(x -> x.maxReceive),
+        Codec.INT.fieldOf("maxExtract").forGetter(x -> x.maxExtract),
+        Codec.INT.fieldOf("energy").forGetter(x -> x.energy)
+    ).apply(i, EnergyStorage::new));
+
+    public static final StreamCodec<FriendlyByteBuf, EnergyStorage> STREAM_CODEC = StreamCodec.composite(
```

Comment:
Generally, I think this should be two stream codecs: one for sending only the energy component and the rest for all the information. Most do not generally need to increase the other parameters on the client after the initial construction.

Reply:
Doesn't most codec-based syncing just replace the object entirely on the client? Dealing with deltas is annoying. :D

Reply:
Yes, but you shouldn't send more information than necessary. For example, you could just sync a single int representing the current energy in 90% of cases.

Reply:
If you just need to sync a single int, then use `ByteBufCodecs` and do it yourself?